play-testkit/src/main/java/play/grpc/testkit/JavaAkkaGrpcClientHelpers.java

```diff
@@ -32,16 +32,20 @@ public static GrpcClientSettings grpcClientSettings(final RunningServer runningS
    */
   public static ServerEndpoint getHttp2Endpoint(final ServerEndpoints serverEndpoints) {
     final scala.collection.Traversable<ServerEndpoint> possibleEndpoints =
-        serverEndpoints.endpoints().filter(func(e->e.expectedHttpVersions().contains("2")));
-    if (possibleEndpoints.size() != 1) {
+        serverEndpoints.endpoints().filter(func(e -> e.expectedHttpVersions().contains("2")));
+    if (possibleEndpoints.size() == 0) {
       throw new IllegalArgumentException(String.format(
           "gRPC client can't automatically find HTTP/2 connection: " +
-              "%s valid endpoints available: %s",
-          possibleEndpoints.size(),
+              "no valid endpoints available. %s",
           serverEndpoints
       ));
+    } else if (possibleEndpoints.size() == 1) {
+      return possibleEndpoints.head();
+    } else {
+      // TODO: the decision on which HTTP/2 endpoint to use should be based on config (e.g. maybe the user set
+      // `akka.grpc.client."".use-tls` to false for gRPC so this should return the non-TLS HTTP/2 endpoint on the list.
+      return possibleEndpoints.filter(endpoint -> endpoint.ssl().isDefined()).head();
```

Comment:
Fail more gracefully if SSL isn't defined for any?

Reply:
Agreed. The correct code here should filter by `HTTP/2` and user's `akka.grpc.client."".use-tls` and crash wonderfully if none is found. 


The next question would be what endpoint to choose if many endpoints fulfill both criteria (protocol version and `use-tls`) if/when we support such a setup.

Reply:
Still, I'd do it on a separate PR.

Reply:
https://github.com/playframework/play-grpc/issues/34

Reply:
I'd prefer if it were done in this PR.  And all I'm looking for is a better exception message if `possibleEndpoints.filter(endpoint -> endpoint.ssl().isDefined())` doesn't have a head element.

Reply:
At the moment, Play doesn't support having two or more `HTTP/2` endpoints where both are `PLAINTEXT` (e.g. none is `encrypted`).

Reply:
The actual issue is returning a `ServerEndpoint` for `HTTP/2+TLS` when the users specified they want their `GrpcClient` to use `use-tls = false`.